src/esplora_client.rs

```diff
@@ -0,0 +1,132 @@
+            .ok()
+            .unwrap_or(None)
+            .map_or(false, |status| status.confirmed)
+    }
+
+    pub fn get_extended_header(
+        &self,
+        block_hash: &BlockHash,
+    ) -> Result<Option<ExtendedBlockHeader>, Error> {
+        if let Some(height) = self.get_height_by_hash(block_hash)? {
+            let header = self.client.get_header_by_hash(block_hash)?;
+
+            return Ok(Some(ExtendedBlockHeader { header, height }));
+        }
+
+        Ok(None)
+    }
+
+    pub fn get_confirmed_tx_by_id(
+        &self,
+        txid: &Txid,
+    ) -> Result<Option<ConfirmedTransaction>, Error> {
+        if let Some(tx_status) = self.client.get_tx_status(txid)? {
+            return self.get_confirmed_tx(txid, &tx_status);
+        }
+
+        Ok(None)
+    }
+
+    pub fn get_confirmed_spending_tx(
+        &self,
+        txid: &Txid,
+        index: u64,
+    ) -> Result<Option<ConfirmedTransaction>, Error> {
+        if let Some(output_status) = self.client.get_output_status(txid, index)? {
+            if output_status.spent {
+                if let (Some(spending_tx_id), Some(spending_tx_status)) =
+                    (output_status.txid, output_status.status)
+                {
+                    return self.get_confirmed_tx(&spending_tx_id, &spending_tx_status);
+                } else {
+                    error!("Esplora sees output {}:{} as spent, yet its spending transaction does not have any id and/or status attributed to it.", txid, index);
+                }
+            }
+        }
+
+        Ok(None)
+    }
+
+    pub fn get_confirmed_tx(
+        &self,
+        txid: &Txid,
+        tx_status: &TxStatus,
+    ) -> Result<Option<ConfirmedTransaction>, Error> {
+        if tx_status.confirmed {
+            if let (Some(block_hash), Some(block_height)) =
+                (tx_status.block_hash, tx_status.block_height)
+            {
+                if let Some(tx) = self.client.get_tx(txid)? {
+                    let block_header = self.client.get_header_by_hash(&block_hash)?;
+                    if let Some(merkle_proof) = self.client.get_merkle_proof(txid)? {
+                        return Ok(Some(ConfirmedTransaction {
+                            tx,
+                            block_height,
+                            block_header,
+                            position: merkle_proof.pos,
+                        }));
+                    } else {
+                        error!("Esplora sees transaction {} as confirmed, but does not have a merkle proof for it", txid);
+                    }
+                } else {
+                    error!(
+                        "Expected transaction {} to be confirmed, but couldn't find it on Esplora",
+                        txid
+                    );
+                }
+            } else {
+                error!("Esplora sees transaction {} as confirmed, yet there is no block hash and/or block height attributed to it.", txid);
```

Comment:
Should we just log and ignore these errors?

Reply:
Yes. These errors point more towards an error within Esplora, than an something going wrong in our app / or some wrong user input. Imo it is sensible to just not let a transaction confirm as long as the requests do not go through. However if these errors happen (unlikely to begin with), they must be reported such that Esplora can be fixed.

In case we don't confirm transactions even though they are confirmed on the blockchain:
- freshly opened channels are not recognized as opened and are therefore not usable (not an issue for 0-conf channels)
- channel closing becomes dangerous. However there is no good way to deal with it if the on-chain sync breaks. External watchtowers should help here.